spanner/spanner_snippets/spanner/integration_test.go

```diff
@@ -604,6 +541,102 @@ func mustRunSample(t *testing.T, f sampleFuncWithContext, dbName, errMsg string)
 	return b.String()
 }
 
+func createTestInstance(t *testing.T) (instanceName string, cleanup func()) {
```

Comment:
Shouldn't we directly pass `projectID` into this function? The current way looks confusing that we parse the projectID from the instanceName and then we construct a new instanceName.

Reply:
What I mean is to move `parseInstanceName` outside of this function to get the projectID first and then call this function.

Reply:
Yes, this is a little confusing / strange in order to keep the tests behave the same as now using the same environment variables. I've moved the instance name parsing outside the `createTestInstance` function to make it a little less confusing (albeit with a little code duplication).